### PR TITLE
fdb7.0: support for grv/commit proxies

### PR DESCRIPTION
**Pull request by ltsampros**
# Description

*Please include a summary of the change and which issue is addressed. If this change resolves an issue, please include the issue number in the description.*

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation
- Other

# Discussion

*Are there any design details that you would like to discuss further?*

# Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

# Documentation

*Did you update relevant documentation within this repository?*

*If this change is adding new functionality, do we need to describe it in our user manual?*

*If this change is adding or removing subreconcilers, have we updated the core technical design doc to reflect that?*

*If this change is adding new safety checks or new potential failure modes, have we documented and how to debug potential issues?*

# Follow-up

*Are there any follow-up issues that we should pursue in the future?*

*Does this introduce new defaults that we should re-evaluate in the future?*

### CODE REVIEW

File: api/v1beta2/foundationdb_database_configuration.go
Comment (ltsampros) [reply api/v1beta2/foundationdb_database_configuration.go]: Added.

File: api/v1beta2/foundationdbcluster_types.go
Comment (ltsampros) [reply api/v1beta2/foundationdbcluster_types.go]: Added.

**Comment by brownleej** [comment]
I still need to review the PR in detail, but I would expect that since FDB itself still supports `proxies` in the status and in the `configure` command, we could continue supporting it in the operator for as long as the user wants to define it.

File: api/v1beta2/foundationdb_database_configuration.go
Comment (johscheuer) [inline api/v1beta2/foundationdb_database_configuration.go:236]: The default number of `LogRouters` will equal the number of `Logs`.

File: api/v1beta2/foundationdb_database_configuration.go
Comment (johscheuer) [inline api/v1beta2/foundationdb_database_configuration.go:524]: Missing the doc string `GetProxiesString ...`

File: api/v1beta2/foundationdb_database_configuration.go
Comment (johscheuer) [inline api/v1beta2/foundationdb_database_configuration.go:532]:
```golang
 	if version.HasSeparatedProxies() && configuration.AreSeparatedProxiesConfigured() {
 		return fmt.Sprintf(" commit_proxies=%d grv_proxies=%d", counts.CommitProxies, counts.GrvProxies)
 	} 

        return fmt.Sprintf(" proxies=%d", counts.Proxies)
```

We can directly return the generated string instead of appending it to the empty variable.

File: api/v1beta2/foundationdbcluster_types_test.go
Comment (johscheuer) [inline api/v1beta2/foundationdbcluster_types_test.go:656]: Could you add some newlines here? So far we always added a newline when we have blocks at the same "level".

File: api/v1beta2/foundationdbcluster_types_test.go
Comment (johscheuer) [inline api/v1beta2/foundationdbcluster_types_test.go:659]: NIT: We should have a constant for the `ssd` storage engine that we could use here.

**Review by johscheuer — CHANGES_REQUESTED** [review]
A few comment otherwise the PR looks good to me. Do you plan to add the user documentation in this PR or another one?

**Review by johscheuer — COMMENTED** [review]
I just noted that the PR checks are not running. I take a look later to see why the checks are not running.

File: api/v1beta2/foundationdb_database_configuration.go
Comment (ltsampros) [reply api/v1beta2/foundationdb_database_configuration.go:236]: That was the original docs of the function. I don't think we need to modify this as the next sentence explains the two cases.

File: api/v1beta2/foundationdbcluster_types_test.go
Comment (ltsampros) [reply api/v1beta2/foundationdbcluster_types_test.go:660]: I'll address this in a separate PR.

File: api/v1beta2/foundationdbcluster_types_test.go
Comment (ltsampros) [reply api/v1beta2/foundationdbcluster_types_test.go:657]: Added

File: api/v1beta2/foundationdb_database_configuration.go
Comment (ltsampros) [reply api/v1beta2/foundationdb_database_configuration.go]: Done.

File: api/v1beta2/foundationdb_database_configuration.go
Comment (ltsampros) [reply api/v1beta2/foundationdb_database_configuration.go:533]: Done.

**Review by johscheuer — APPROVED** [review]
LGTM 👍 once the tests pass we can merge it. Please open an issue to document the role counts and how they behave for cluster pre `7.0` and `7.0+`

File: api/v1beta2/foundationdb_database_configuration.go
Comment (brownleej) [inline api/v1beta2/foundationdb_database_configuration.go:518]: Is this equivalent to `return counts.Proxies == 0 && (counts.GrvProxies > 0 && counts.CommitProxies > 0)`?

**Review by brownleej — APPROVED** [review]
👍

**Pull request merged**
